pkg/server/structlogging/hot_ranges_log_test.go

```diff
@@ -101,72 +90,86 @@ func TestHotRangesStats(t *testing.T) {
+// The test will ensure each node contains 5 distinct range replicas for hot
+// ranges logging. Each node should thus log 5 distinct range ids.
+func TestHotRangesStats(t *testing.T) {
+	skip.UnderRace(t)
+	ctx := context.Background()
+	ts, spy, teardown := setupHotRangesLogTest(t, ctx)
+	defer teardown()
+
+	structlogging.TelemetryHotRangesStatsEnabled.Override(ctx, &ts.ClusterSettings().SV, true)
+	structlogging.TelemetryHotRangesStatsInterval.Override(ctx, &ts.ClusterSettings().SV, time.Millisecond)
+	structlogging.TelemetryHotRangesStatsLoggingDelay.Override(ctx, &ts.ClusterSettings().SV, 0*time.Millisecond)
+
+	structlogging.HotRangeLogManualTicker <- struct{}{}
+	testutils.SucceedsSoon(t, func() error {
+		logs := spy.Logs()
 
-		// Get first 5 logs since the logging loop may have fired multiple times.
-		// We should have gotten 5 distinct range ids, one for each split point above.
-		logs := spy.Logs()[:5]
+		// Depend on a range which we don't exist to go anywhere.
 		for _, l := range logs {
-			assert.Equal(t, l.Databases, []string{"‹test›"})
-			assert.Equal(t, l.Tables, []string{"‹foo›"})
-			assert.Equal(t, l.Indexes, []string{"‹foo_pkey›"})
-			_, ok := rangeIDs[l.RangeID]
-			if ok {
-				t.Fatalf(`Logged ranges should be unique per node for this test.
-found range on node %d and node %d: %s %s %s %s %d`, i, l.LeaseholderNodeID, l.Databases, l.SchemaName, l.Tables, l.Indexes, l.RangeID)
+			if !slices.Equal(l.Databases, []string{"‹system›"}) {
+				continue
 			}
-			rangeIDs[l.RangeID] = struct{}{}
+			if !slices.Equal(l.Tables, []string{"‹sqlliveness›"}) {
+				continue
+			}
+			if !slices.Equal(l.Indexes, []string{"‹primary›"}) {
+				continue
+			}
+			return nil
 		}
+		return errors.New("waited too long for the synthetic data")
+	})
+}
 
+func TestHotRangeLogIntervalSetting(t *testing.T) {
+	skip.UnderRace(t)
+	ctx := context.Background()
+	ts, spy, teardown := setupHotRangesLogTest(t, ctx)
+	defer teardown()
+
+	intervalDuration := 50 * time.Millisecond
+	start := time.Now()
+	structlogging.TelemetryHotRangesStatsEnabled.Override(ctx, &ts.ClusterSettings().SV, true)
+	structlogging.TelemetryHotRangesStatsInterval.Override(ctx, &ts.ClusterSettings().SV, intervalDuration)
+	structlogging.TelemetryHotRangesStatsLoggingDelay.Override(ctx, &ts.ClusterSettings().SV, 0*time.Millisecond)
+
+	// very that there's no logged hot ranges, despite the system ticking
+	structlogging.HotRangeLogManualTicker <- struct{}{}
+	if time.Since(start) > intervalDuration {
+		// stress tests can cause scheduling delays, so that little code takes a
+		// long time to run. we can bail if the above took longer than the expected
+		// duration.
+		return
 	}
+
+	// verify no logs were sent.
+	assert.Zero(t, len(spy.Logs()))
+
+	// sleep for the duration, retrigger the logger and verify that logs were sent.
+	time.Sleep(intervalDuration * 2)
+	testutils.SucceedsSoon(t, func() error {
```

Comment:
Typically we discourage `test.Sleep` to wait out intervals. You're using `SucceedsSoon` already so there's no need to sleep.

Reply:
Makes sense, I'll get rid of this one.